src/main/java/bio/terra/app/controller/ApiValidationExceptionHandler.java

```diff
@@ -66,6 +69,25 @@
         return new ResponseEntity<>(errorModel, HttpStatus.BAD_REQUEST);
     }
 
+    @Override
+    protected ResponseEntity<Object> handleTypeMismatch(
+        TypeMismatchException ex, HttpHeaders headers, HttpStatus status, WebRequest request) {
+        String message = status + " - see error details";
+        List<String> details = Collections.emptyList();
+        for (Throwable cause = ex.getCause(); cause != null; cause = cause.getCause()) {
+            if (cause instanceof DataRepoException) {
+                message = cause.getMessage();
+                details = ((DataRepoException) cause).getErrorDetails();
+            } else {
+                details = Collections.singletonList(cause.getMessage());
+            }
+        }
+        ErrorModel errorModel = new ErrorModel().message(message)
```

Comment:
The error response won't include `ex.getMessage()`, is that intentional?

Reply:
Yes. The `ValidationException` being thrown isn't actually what's caught. Its wrapped up in several Spring exceptions that need to be unwrapped to get to our root exception. We don't care about the messages from the wrapping exceptions since they're not all that helpful to the user.